### ISSUE TITLE
vectorized string function naming scheme

### ISSUE THREAD

**Issue by kwmsmith** [blaze/blaze#1532]
I like the `t.col.str.upper()` API (and likewise for datetime, with `t.col.dt.zzz()`).

I'm not wild about our current string function naming scheme:
* `str_find`
* `str_len`
* `str_upper`
* etc.

I'd much rather we recommend something like:

```python
from blaze import string as bstr

bstr.len(t.col) == t.col.str.len()
bstr.upper(t.col) == t.col.upper()
# etc...
```

That way the function-level API is much more streamlined, and we let the user determine how they are to be namespaced.

@llllllllll @stefanseefeld Thoughts?

**Comment by stefanseefeld** [comment]
I fully agree. I find the 'str_' prefix quite redundant, given that these symbols already come from the 'strings' module.

**Comment by llllllllll** [comment]
Is this just for the functional form of the expressions? I think the `expr.str.*` is nice for methods but I would support this for the functional versions.

**Comment by kwmsmith** [comment]
> Is this just for the functional form of the expressions?

Yes, exactly -- the method form would remain `expr.str.zzz()`, which would be equivalent to `bstr.zzz(expr)`.

**Comment by stefanseefeld** [comment]
Do we need to support the existing API (and deprecate it) ?

**Comment by stefanseefeld** [comment]
This is being addressed by https://github.com/blaze/blaze/pull/1538.

Note that there are two distinct types of redundancy:

* The naming of functions in the blaze.expr.strings module.
* The appearance of certain methods both within the `str` namespace as well as outside it. (Example: the `s.name.str.len()` method can also be spelled `s.name.str_len()`. The latter is now implicitly deprecated by virtue of the stand-alone `str_len()` function being deprecated.

**Comment by kwmsmith** [comment]
Closed by #1538.
